### PR TITLE
rcanary cleanup, "is fixed" check

### PR DESCRIPTION
**Pull request by gyng**
- Update dependencies
- Change string statuses to an enum
- Move alert mail sending into a thread, might fix random hangs
- Add "is fixed" functionality
- Make tests more readable
- Change test server listening address to 127.0.0.1 (fixes test failure on Windows)

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -65,13 +65,20 @@ struct CanaryTarget {
 #[derive(RustcEncodable, Eq, PartialEq, Clone, Debug)]
 pub struct CanaryCheck {
     target: CanaryTarget,
-    info: String,
+    info: Status,
```
Comment (lawliet89) [inline src/main.rs]: Maybe we should rename this to `status` in the future.

File: src/main.rs [hunk 2]
```diff
@@ -120,10 +127,15 @@ fn main() {
...
 
-        let is_spam = check_spam(&mut last_statuses, &result);
+        let is_spam = check_spam(&last_statuses, &result);
+        let is_fixed = check_fixed(&last_statuses, &result);
+        last_statuses.insert(result.target.clone(), result.info.clone());
 
-        if config.alert.enabled && result.alert && result.need_to_alert && !is_spam {
-            let _ = send_alert(&config.alert, &result);
+        if config.alert.enabled && result.alert && (is_fixed || result.need_to_alert && !is_spam) {
+            println!("Sending alert for {:?}", result);
```
Comment (lawliet89) [inline src/main.rs]: Probably should log to the same place as `log_result`.

File: src/main.rs [hunk 3]
```diff
@@ -372,151 +412,90 @@ mod tests {
 
     #[test]
     fn it_marks_as_spam_on_empty_history() {
-        let target = CanaryTarget {
```
Comment (lawliet89) [inline src/main.rs]: 😥

**Comment by gyng** [comment]
Note to self: check if spam checking is prevent event emission

**Pull request merged**